### ISSUE TITLE
record can not be used in submit, rectype sent instead of file data

### ISSUE THREAD

**Issue by tazzix** [voiceglue/voiceglue#10]
According to samples a submit containing a record name in its namelist with post and multipart/form-data should POST the the recorded data. Instead it sends rectype, data length and a binary address of some sort:

09:35:49:565 DBUG OPEN_VXI svr001-- callid=[412] SBinetChannel::open(http://localhost/vxml/sub.php) called with mode FILE and with properties: {inet.submitMethod=POST inet.submitMimeType=multipart/form-data inet.timeoutDownload=7000 inet.urlBase=http://localhost/vxml/yasir.php inet.urlQueryArgs={abc=audio/basic:len(40000):p0x2aaaac1c0c00}}

Moreover the following CURL command being called causes a wrong enctype as well, obvious by the usage of "-d" parameter:

09:35:49:571 DBUG HTTCACHE svr001-- hcreq=[1010] started pid=25606 as curl -k -m 7 -q -f -s -S --create-dirs -o /var/lib/asterisk/sounds/voiceglue/url/SqKeVxk_9E45hixyjTq2A-s8F81wQv5U1_sub.php/req.tmp -b /var/lib/asterisk/sounds/voiceglue/cookie/412 -c /var/lib/asterisk/sounds/voiceglue/cookie/412 -D /var/lib/asterisk/sounds/voiceglue/url/SqKeVxk_9E45hixyjTq2A-s8F81wQv5U1_sub.php/req.header --stderr /var/lib/asterisk/sounds/voiceglue/url/SqKeVxk_9E45hixyjTq2A-s8F81wQv5U1_sub.php/req.err -d abc=audio/basic:len(40000):p0x2aaaac1c0c00 -b /var/lib/asterisk/sounds/voiceglue/cookie/412 http://localhost/vxml/sub.php

For reference my XML:

<?xml version="1.0"?>
<vxml version="2.0">
    <form>
        <record name="myrecording" beep="true">
            <prompt>
                Please record a message after the beep.
            </prompt>

```
        <filled>
            <submit next="submitrecording.php" namelist="myrecording"
            method="post" enctype="multipart/form-data"/>
        </filled>
    </record>
</form>
```

</vxml>
# 

submitrecording.php:
<?php
header("Content-type: text/xml");
echo("<?xml version=\"1.0\"?>\n");
?>
<vxml version="2.0">
  <form>
    <block>

<?php
if (isset($_FILES['myrecording']) && is_uploaded_file($_FILES['myrecording']['tmp_name'])) {
        move_uploaded_file($_FILES['myrecording']['tmp_name'],"message.wav");
        echo "<prompt bargein=\"false\">Audio saved.</prompt>";
}  else {
        echo "<prompt bargein=\"false\">Audio not saved.</prompt>";
}
?>

```
</block>
```

  </form>
</vxml>

**Comment by tazzix** [comment]
request for comments!!

**Comment by arnaudmorin** [comment]
Hi, 
We are facing exactly the same issue here. Any news?

Cheers

**Comment by tazzix** [comment]
no, and the project seems to be dead anyways. no pushes in 9 months :p

**Comment by voiceglue** [comment]
Hi guys, the project isn't dead, we just have a long release cycle, and tend to do a fairly significant regression test before major releases.  One of the challenges with this project is that it involves a lot of different open source components, and while we do look at and fix openVxi and other subsystem issues, we can't always dig into them in a timely fashion.

**Comment by tazzix** [comment]
no offense, but it seems the record thing was not even unit or smoke tested, let alone regression tested. There is no fairly simple way of playing back the recording or even get the filename to do anything with it at all.

**Comment by voiceglue** [comment]
OK I chatted this through internally and we agree that we need to look at this and get a fix out.   Doug can spend some time on this the week after next.  His first step will be to verify the problem, it sounds like it happens with any recording submit so it should be easy to recreate.    We'll then see what the issue is and how to fix it, and provide a push or at the minimum an update to this thread.    It sounds like we also need to update our regression test procedures for this case, so thanks for helping point that out (and no offense taken :).

**Comment by voiceglue** [comment]
p.s. we believe that this did work in 0.12, so it sounds like it perhaps was a regression; in which case it hopefully won't be too hard to track down.

**Comment by voiceglue** [comment]
we're working on this today, stay tuned for info; we'll at least post an update as to how far we got

**Comment by voiceglue** [comment]
I've been able to replicate the issue.  It does appear to have regressed, since it definitely did work in the 0.12 branch.  A fix is being worked on.

**Comment by arnaudmorin** [comment]
Great news! Waiting for the fix.

**Comment by arnaudmorinol** [comment]
As far as I can see, no fix has been provided. Do you plan to commit this fix one day? 
Did the project move to any other source code repository?

**Comment by amr132** [comment]
Is there any update about the fix
